theano/sandbox/cuda/opt.py

```diff
@@ -1814,6 +1814,12 @@ def split_huge_add_or_mul(node):
 optdb.register('gpu_inplace_elemwise_opt', gpu_inplace_elemwise_optimizer, 75,
                'fast_run', 'inplace', 'gpu_inplace')
 
+tensor.opt.register_specialize_device(tensor.opt.local_shape_to_shape_i)
```

Comment:
I think both opt should be registered only in register_opt. If you do that, do it work?